src/main/java/org/broadinstitute/hellbender/tools/copynumber/utils/UnionSegments.java

```diff
@@ -0,0 +1,195 @@
+            fullName = COLUMNS_OF_INTEREST_LONG_NAME, shortName = COLUMNS_OF_INTEREST_SHORT_NAME, minElements = 1
+    )
+    protected Set<String> columnsOfInterest = new HashSet<>();
+
+    @Argument(
+            doc="Output tsv file with union'ed segments",
+            shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME)
+    protected File outputFile;
+
+    @Override
+    public void traverse() {
+        try {
+
+            final VersatileAnnotatedRegionParser parser = new VersatileAnnotatedRegionParser();
+            final List<SimpleAnnotatedGenomicRegion> segments1 = parser.readAnnotatedRegions(segmentsFile.get(0), columnsOfInterest);
+            final List<SimpleAnnotatedGenomicRegion> segments2 = parser.readAnnotatedRegions(segmentsFile.get(1), columnsOfInterest);
+            final List<List<SimpleAnnotatedGenomicRegion>> inputSegmentsLists = Arrays.asList(segments1,segments2);
+
+            if (segmentsFileLabels.size() == 0) {
+                segmentsFileLabels = IntStream.range(0, inputSegmentsLists.size()).mapToObj(i -> String.valueOf(i+1)).collect(Collectors.toList());
+            }
+
+            // Check to see if we should warn the user that one or more columns of interest were not seen in any input file.
+            final Set<String> allSeenAnnotations = Sets.union(segments1.get(0).getAnnotations().keySet(), segments2.get(0).getAnnotations().keySet());
+            final Set<String> unusedColumnsOfInterest = Sets.difference(columnsOfInterest, allSeenAnnotations);
+            if (unusedColumnsOfInterest.size() > 0) {
+                final List<String> missingColumns = new ArrayList<>(unusedColumnsOfInterest);
+                logger.warn("Some columns of interest specified by the user were not seen in any input files: " + StringUtils.join(missingColumns, ", "));
+            }
+
+            // Create a map of input to output headers.  I.e. the annotation in the segment1 to the output that should be written in the final file.
+            //  This assumes that the keys in each entry of the list is the same.
+            // TODO: If we want to support more than two segment files, this is the only bit that requires thinking.  Once this is solved, then this logic can go into a utility class.
+            final Set<String> intersectingAnnotations = Sets.intersection(segments1.get(0).getAnnotations().keySet(), segments2.get(0).getAnnotations().keySet());
+
+            // Create the obvious mappings that are identity then tack on new annotations for conflicts.
+            // These are mappings that take the header name from the segment files and map to an output header to avoid conflicts.
+            final Map<String, String> input1ToOutputHeaderMap = Utils.stream(segments1.get(0).getAnnotations().keySet().iterator()).filter(a -> !intersectingAnnotations.contains(a))
```

Comment:
Why do you need to use `Utils.stream` here and elsewhere?

Reply:
You can't stream a set.  I'm open to suggestions.

No action.

Reply:
You definitely can, but there are no guarantees on order unless the implementation guarantees it (e.g., LinkedHashSet/LinkedHashMap).  I suggest using such an implementation to guarantee the column order specified in the command line (as opposed to that in the files).  Add the corresponding test for good measure.